demo-apps/cf-plugtest-client/src/main/java/org/eclipse/californium/plugtests/PlugtestClient.java

```diff
@@ -107,6 +115,9 @@ public void applyDefinitions(Configuration config) {
 	 */
 	public static void main(String[] args) throws ConnectorException, IOException {
 
+		HashMapCtxDB db = new HashMapCtxDB();
```

Comment:
Please consider to use a cli-option (default enabled) to switch OSCORE on/off.

Reply:
I struggle a bit in how to do this in the best way. The command line parameters are parsed in `ClientInitializer.init` and it is also in there the Endpoint is created.

However I need to know whether to do `OSCoreCoapStackFactory.useAsDefault(db);` before the endpoint is created, which I can't know since I don't know the results of the command line argument parsing.

For alternatives I came up with three options

- Adding the cf-oscore as a dependency to cf-cli, creating the HashMapCtxDB and setting it with `OSCoreCoapStackFactory.useAsDefault(db);` inside there (using `setCustomCoapStackArgument(customStackArgument)` & `setCoapStackFactory` should work too).
- Moving out the Endpoint creation from `ClientInitializer.init` into the code of the plugtest client.
- Manually pre-parsing the command line options (for OSCORE only). If OSCORE is used I can add to the clientConfig the OSCORE context DB as customStackArgument and the OSCoreCoapStackFactory as coapStackFactory. Then within `ClientInitializer.init` if OSCORE is used those values are read from the config and used when creating the Endpoint.

Not sure what you prefer or if there are better options.

Reply:
With PR #1972 it will be

```
		ClientInitializer.init(args, clientConfig, false);
		HashMapCtxDB db = null;
		if (clientConfig.oscore) {
			db = new HashMapCtxDB();
			OSCoreCoapStackFactory.useAsDefault(db);
		}
		ClientInitializer.registerEndpoint(clientConfig, null);
```

Reply:
PR #1972 is merged. If you rebase, you should be able to apply to above snippet.

Reply:
Thanks!  Now I have added cli-option for the client in commit 20aa762ebf9b41e73f67f4b9fd28afdf2f08ebd0